pysolr.py

```diff
@@ -823,7 +825,17 @@ def delete(self, id=None, q=None, commit=True, waitFlush=None, waitSearcher=None
         elif id is not None and q is not None:
             raise ValueError('You many only specify "id" OR "q", not both.')
         elif id is not None:
-            m = '<delete><id>%s</id></delete>' % id
+            if type(id) not in (list, set):
+                id = [ id ]
+            else:
+                id = filter(lambda x: x is not None, id)
```

Comment:
This should be shortened to `filter(None, id)` which would also catch empty strings:

https://docs.python.org/3/library/functions.html#filter